js/battle.js

```diff
@@ -3380,6 +3380,7 @@ var Battle = (function () {
 					switch (effect.id) {
 					default:
 						if (effect.effectType === 'Ability') {
+							fromAbility = true;
 							this.resultAnim(poke, effect.name, 'ability', animDelay);
```

Comment:
An alternative approach is to reset animDelay to 2 here so that it's always correct later on when you need it.